runtime/stdlib/contracts/test.cdc

```diff
@@ -250,4 +258,49 @@ pub contract Test {
         ///
         pub fun useConfiguration(_ configuration: Configuration)
     }
+
+    /// Returns a new matcher that checks if the given test value is either
+    /// a ScriptResult or TransactionResult and the ResultStatus is succeeded.
+    /// Returns false in any other case.
+    ///
+    pub fun beSucceeded(): Matcher {
+        return Matcher(test: fun (value: AnyStruct): Bool {
+            if value.getType() == Type<TransactionResult>() {
+                let result = value as! TransactionResult
+                return result.status == ResultStatus.succeeded
+            } else if value.getType() == Type<ScriptResult>() {
+                let result = value as! ScriptResult
+                return result.status == ResultStatus.succeeded
+            } else {
+                return false
+            }
+        })
+    }
+
+    /// Returns a new matcher that checks if the given test value is either
+    /// a ScriptResult or TransactionResult and the ResultStatus is failed.
+    /// Returns false in any other case.
+    ///
+    pub fun beFailed(): Matcher {
+        return Matcher(test: fun (value: AnyStruct): Bool {
+            if value.getType() == Type<TransactionResult>() {
```

Comment:
Same as above, use failable downcasting (`as?`) instead

Reply:
Fixed in https://github.com/onflow/cadence/pull/2420/commits/27bed60c70d2e52ee2aa0c4fdd60df8c03a993db